### ISSUE TITLE
PartialView is cahced somehow

### ISSUE THREAD

**Issue by nefcanto** [aspnet/Mvc#7625]
I deployed an ASP.NET Core MVC application to my server, and hosted it in IIS. Configured binding and everything is working. Here's the problem:

- I change a cshtml file
- I only upload that cshtml file onto the server, inside the relevant folder
- I call the public action, that returns the PartialView of that cshtml file
- It's not the updated version
- I delete cshtml file from server, and call that action again
- It serves the file, while file is not there anymore physically
- I restart IIS, again file is served
- I kill dotnet.exe, again file is served
- I completely remove Views folder, still file is served

It seems that ASP.NET Core MVC caches the `PartialView` response somehow. This is good of course for performance, but not when I want to get a fresh version. The response from IIS is 200, and there is no caching response in IIS.

What can be wrong here?

**Comment by khellang** [comment]
Sounds like you're shipping precompiled views. See https://docs.microsoft.com/en-us/aspnet/core/mvc/views/view-compilation?tabs=aspnetcore2x. The default behavior varies a bit depending on the version of ASP.NET Core you're running.

Do you have a *.PrecompiledViews.dll file in your apps bin folder?

**Comment by nefcanto** [comment]
No I haven't use precompiled views, and I HAVE NOT configured my project file this way:

```
  <ItemGroup>
    <EmbeddedResource Include="Views\**\*.cshtml" />
  </ItemGroup>
```

Yet for some weird reason I see a *.PrecompiledViews.dll in my *obj* folder, not in my *bin* folder.

**Comment by nefcanto** [comment]
Yeah, you're right, thought *.PrecompiledViews.dll is not in my **bin** folder, it's present in my published folder on my server. This is strange, because I can see that we have not configured csproj file to do the view precompilation.

**Comment by khellang** [comment]
What version of ASP.NET Core are you running? If you're running 2.0, precompiled views are turned on by default:

> The ASP.NET Core 2.x project templates implicitly set MvcRazorCompileOnPublish to true by default, which means this node can be safely removed from the .csproj file.

If you're targeting 2.0 and have `<Project Sdk="Microsoft.NET.Sdk.Web">` in the top of your project file, you're probably publishing precompiled views without knowing it.

> Yet for some weird reason I see a *.PrecompiledViews.dll in my obj folder, not in my bin folder.

> Yeah, you're right, thought *.PrecompiledViews.dll is not in my bin folder, it's present in my published folder on my server.

Yeah, by bin folder, I was really referring to your published application folder 😄

**Comment by nefcanto** [comment]
I'm using `<PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.5" />` and I do have `<Project Sdk="Microsoft.NET.Sdk.Web">` at the top of my project.

**Comment by mkArtakMSFT** [comment]
@pranavkm, any suggestions here? Thanks!
